test/python/qasm3/test_export.py

```diff
@@ -2665,6 +2665,40 @@ def test_switch_v1_expr_target(self):
         test = dumps(qc, experimental=ExperimentalFeatures.SWITCH_CASE_V1)
         self.assertEqual(test, expected)
 
+    def test_global_barrier_all_qubits(self):
+        """Simple test that the global barrir is created"""
+        qc = QuantumCircuit(3)
+        qc.x(0)
+        qc.barrier()
+        qc.x(1)
+        expected = """\
+OPENQASM 3.0;
+include "stdgates.inc";
+qubit[3] q;
+x q[0];
+barrier;
+x q[1];
+"""
+        test = dumps(qc, experimental=ExperimentalFeatures.GLOBAL_BARRIER)
+        self.assertEqual(test, expected)
+
+    def test_global_barrier_partial_qubits(self):
+        """Simple test that the global barrir is created"""
+        qc = QuantumCircuit(3)
+        qc.x(0)
+        qc.barrier(0, 1)
+        qc.x(1)
+        expected = """\
+OPENQASM 3.0;
+include "stdgates.inc";
+qubit[3] q;
+x q[0];
+barrier q[0], q[1];
+x q[1];
+"""
+        test = dumps(qc, experimental=ExperimentalFeatures.GLOBAL_BARRIER)
+        self.assertEqual(test, expected)
+
```

Comment:
Perhaps we could combine the two programs into a single OpenQASM 3 string, so we test both at once, and make sure we have coverage of when there's more than one barrier in a circuit?

Could you also add a test that the handling works when there's a `barrier` inside a control-flow statement, and the `barrier` affects all the qubits touched by the control-flow operation, but _not_ all the qubits in the circuit?  In this case, we mustn't write out a global barrier.